### DISCUSSION TITLE
Curso node | Criando APIS com node | Aula: Excluindo um produto

### DISCUSSION THREAD

**Discussion by MalinaliRap** [balta-io/forum#359] [Tira Dúvidas] [answered]
Estou com uma dúvida, na aula que eu cito no titulo, é utilizado `findOneAndRemove(req.body.id)` para remover o produto em questão, porém se eu colocar outro id que não seja do produto, ou qualquer outra coisa no lugar ele está excluindo todos os produtos, segue abaixo a função do controller e a rota:

### Controller - Função Del

`exports.del = (req, res, next) => {
  Product.findOneAndRemove(req.body.id)
    .then((x) => {
      res.status(200).send({
        message: "Produto removido com sucesso",
      })
    })
    .catch((e) => {
      res.status(400).send({
        message: "Falha ao remover produto",
        data: e,
      })
    })
}`

### Arquivo de rotas

`router.delete("/", controller.del)`

### Chamada pelo postman

`DELETE http://localhost:3000/products`

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @MalinaliRap como vai?

Consegue garantir que ele está realmente chamando a função `findOneAndRemove`? É um comportamento bem estranho.
Se puder debugar e ver se ele realmente chega nesta rota/linha quando não informa nada ou algum outro valor como Id por favor.

Se este for o comportamento padrão do Mongoose, não tem muito o que a gente fazer... no máximo verificar se o Id é realmente um Id e se não é vazio.
